src/main/java/uk/gov/ons/census/fwmt/rmadapter/service/impl/RMAdapterServiceImpl.java

```diff
@@ -37,26 +39,43 @@ public void sendJobRequest(ActionInstruction actionInstruction) throws GatewayEx
       createUpdateMessage(actionInstruction);
     } else {
       String msg = "No matching request was found";
-      gatewayEventManager.triggerErrorEvent(this.getClass(), msg, actionInstruction.getActionRequest().getCaseId(), INVALID_ACTION_INSTRUCTION);
+      String unknown = "Unknown caseId";
+      gatewayEventManager.triggerErrorEvent(this.getClass(), msg, unknown, INVALID_ACTION_INSTRUCTION);
       throw new GatewayException(GatewayException.Fault.SYSTEM_ERROR, msg);
     }
   }
 
   private void createUpdateMessage(ActionInstruction actionInstruction) throws GatewayException {
-    jobServiceProducer.sendMessage(CanonicalJobHelper.newUpdateJob(actionInstruction));
-    gatewayEventManager.triggerEvent(actionInstruction.getActionUpdate().getCaseId(), CANONICAL_UPDATE_SENT);
+    if (householdStore.retrieveCache(actionInstruction.getActionUpdate().getCaseId()) != null) {
+      if (actionInstruction.getActionUpdate().getAddressType().equals("HH")) {
+        jobServiceProducer.sendMessage(CanonicalJobHelper.newUpdateJob(actionInstruction));
+        gatewayEventManager.triggerEvent(actionInstruction.getActionUpdate().getCaseId(), CANONICAL_UPDATE_SENT);
+      } else {
+        throw new GatewayException(GatewayException.Fault.SYSTEM_ERROR, "Valid address type not found");
+      }
+    } else {
```

Comment:
remove the else, let them die silently, actually the surrounding if statement could read
if (householdStore.retrieveCache(actionInstruction.getActionUpdate().getCaseId()) == null)  return;

Reply:
As discussed, this has been altered.